### PR TITLE
Fix crash from west to east

### PR DESCRIPTION
**Pull request by ThanatosGit**
Please test it also. Thx

### CODE REVIEW

**Comment by dyceron** [comment]
Still crashes
```[ 237.689095] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x0000000E at PC 0x00148034
[ 237.689098] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000012 at PC 0x00148034
[ 237.689147] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x0000000E at PC 0x0019DC20
[ 237.689148] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000012 at PC 0x0019DC20
[ 237.689149] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000011 at PC 0x0014803C
[ 237.689156] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x0000000E at PC 0x00148684
[ 237.689157] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000012 at PC 0x00148684
[ 237.689157] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x0000000E at PC 0x0019DC08
[ 237.689158] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000012 at PC 0x0019DC08
[ 237.689279] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x0000000E at PC 0x0014830C
[ 237.689280] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000012 at PC 0x0014830C
[ 237.689281] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x0000000E at PC 0x0019DC08
[ 237.689281] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000012 at PC 0x0019DC08
[ 237.689454] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000011 at PC 0x0016607C
[ 237.689564] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x00000015 at PC 0x001483B0
[ 237.689566] HW.Memory <Error> core/memory.cpp:Write:480: unmapped Write32 0xFFFFFFFF @ 0x00000015 at PC 0x001483B0
[ 237.755363] HW.Memory <Error> core/memory.cpp:Read:434: unmapped Read32 @ 0x6E64BC38 at PC 0x005CC5D0

**Comment by ThanatosGit** [comment]
Then I need the patcher file because it wasn't crashing anymore with my test.

**Comment by dyceron** [comment]
I did multiple tests and so far I haven't ran into any issues. Morphing also doesn't throw errors for me. Seems to be fixed?

**Comment by ThanatosGit** [comment]
Hmmm haven't tocuhed anything after it throws errors in morph state. At least we should fix the oob stuff. You are just a little bit oob on the tiny slope and then fall through the floor after a few moments.

**Comment by dyceron** [comment]
I think the issue is that Samus gets unmorphed after loading a new scenario, so it just starts at the height of the Morph Ball which ends up being too low to stand.

**Comment by ThanatosGit** [comment]
There is a function called something like ForceToSamus which should do the job if we call it before loading the next scenario

**Comment by ThanatosGit** [comment]
Should be good now.

File: src/open_samus_returns_rando/files/custom_scenario.lua [hunk 1]
```diff
@@ -81,4 +81,12 @@ function Scenario.ShowText()
...
     GUI.LaunchMessage("#RANDO_STARTING_TEXT_" .. Init.tBoxesSeen, "Scenario.ShowText", "")
   end
+end
+
+function Scenario.LoadNewScenario(target_scenario, target_spawnpoint)
+  bWaitingForScenarioChange = true
+  Game.FadeOut(0.0)
+  Game.FadeOutStream(0.0 * 0.8)
+  Game.AddPSF(0.1, "Game.LoadScenario", "ssssi", "c10_samus", target_scenario, target_spawnpoint, "", 1)
+  Game.GetPlayer().vPos = V3D(-22800, 4450, 0)
```
Comment (dyceron) [inline src/open_samus_returns_rando/files/custom_scenario.lua:91]: This probably shouldn't be included in the function if it's also being used for warp to start

File: src/open_samus_returns_rando/files/custom_scenario.lua [hunk 2]
```diff
@@ -81,4 +81,12 @@ function Scenario.ShowText()
     Init.tBoxesSeen = textboxes_seen + 1
     GUI.LaunchMessage("#RANDO_STARTING_TEXT_" .. Init.tBoxesSeen, "Scenario.ShowText", "")
   end
+end
+
+function Scenario.LoadNewScenario(target_scenario, target_spawnpoint)
+  bWaitingForScenarioChange = true
+  Game.FadeOut(0.0)
+  Game.FadeOutStream(0.0 * 0.8)
```
Comment (dyceron) [inline src/open_samus_returns_rando/files/custom_scenario.lua:89]: Since this is explicit, can this be simplified?

**Review by dyceron — CHANGES_REQUESTED** [review]

File: src/open_samus_returns_rando/files/custom_scenario.lua (hunk 2 above)
Comment (ThanatosGit) [reply src/open_samus_returns_rando/files/custom_scenario.lua]: Yeah, I just replaced the argument with 0.0 and didn't look at it. Changed.

File: src/open_samus_returns_rando/files/custom_scenario.lua (hunk 1 above)
Comment (ThanatosGit) [reply src/open_samus_returns_rando/files/custom_scenario.lua]: Oops. Doesn't change anyting logically but bad to have it there.
Removed.

**Review by dyceron — APPROVED** [review]

**Pull request merged**